File: selection_sort.cpp
// Time Complexity: O(n^2)
// Space Complexity: O(1)

#include<iostream>

using namespace std;

void selectsort(int [], int);

int main()
{	
	int m[50],n,i;
	cout<<"Hom many Elements , you want to enter:\n";
	cin>>n;
	cout<<"\nEnter the elements:\n";
	for( i=0; i<n; i++ )
	{	
		cin>>m[i];
	}

	selectsort(m,n);
	cout<<"\nSorted Array:\n";
	for( i=0; i<n; i++)
	{	
		cout<<m[i]<<"\t";
	}
	return 0;
}

void selectsort(int p[], int size)
{
	int min,pos,temp;
	for(int t=0; t<(size-1); t++)
	{	min=p[t];
		pos=t;

		for( int j=t+1; j<size; j++)
		{
			if(p[j]<min)
			{	
				min=p[j];
				pos=j;
			}
			
		}

		temp=p[t];
		p[t]=p[pos];
		p[pos]=temp;
	}
}

File: Sorting/selectionsort.java
package git;
import java.util.*;
// time complexity = O(n^2)
// space complexity = O(1)

public class selectionsort {

	    void selectionsort(int a[])
	    {
	        int n = a.length;
	 
	        for (int i = 0; i < n-1; i++)
	        {
	            
	            int min= i;
	            for (int j = i+1; j < n; j++)
	                if (a[j] < a[min])
	                    min = j;
	 
	           
	            int temp = a[min];
	            a[min] = a[i];
	            a[i] = temp;
	        }
	    }
	 
	    void output(int a[])
	    {
	        int n = a.length;
	        for (int i=0; i<n; ++i)
	            System.out.print(a[i]+" ");
	        System.out.println();
	    }
	
	    public static void main(String args[])
	    {
	        selectionsort ob = new selectionsort();
	        Scanner input = new Scanner(System.in);
	        System.out.println("Enter the length array you want to be sorted"); 
	        int n;
	        n=input.nextInt();
			System.out.println("Enter the array(6) you want to be sorted");
			int i;
			int a[] = new int[n];
			for( i=0;i<n;i++)
			{
	        a[i] = input.nextInt();  //to take the input for array members
			}
	        ob.selectionsort(a);
	        System.out.println("ARRAY SORTED USIMG SELECTION SORT ");
	        ob.output(a);
	    }
	}
